x-pack/plugins/remote_clusters/public/application/services/ui_metric.ts

```diff
@@ -5,15 +5,20 @@
  */
 
 import { UsageCollectionSetup } from 'src/plugins/usage_collection/public';
-import { UiStatsMetricType } from '@kbn/analytics';
 import { UIM_APP_NAME } from '../constants';
 
-export let trackUiMetric: (metricType: UiStatsMetricType, eventName: string) => void;
-export let METRIC_TYPE: UsageCollectionSetup['METRIC_TYPE'];
+export let usageCollection: UsageCollectionSetup | undefined;
 
-export function init(usageCollection: UsageCollectionSetup): void {
-  trackUiMetric = usageCollection.reportUiStats.bind(usageCollection, UIM_APP_NAME);
-  METRIC_TYPE = usageCollection.METRIC_TYPE;
+export function init(_usageCollection: UsageCollectionSetup): void {
+  usageCollection = _usageCollection;
+}
+
+export function trackUiMetric(type: 'COUNT' | 'CLICK' | 'LOADED', name: string) {
```

Comment:
👍 Great suggestion. I will update the PR.

Reply:
After looking into this some more, I don't believe your suggestion works, as `UiStatsMetricType` is actually a type that references the enum.

```
export type UiStatsMetricType = METRIC_TYPE.CLICK | METRIC_TYPE.LOADED | METRIC_TYPE.COUNT;
```

So, `trackUiMetric('count', eventName);` results in `Argument of type '"count"' is not assignable to parameter of type 'UiStatsMetricType'.`

I ended up importing `METRIC_TYPE` from `@kbn/analytics`. I think this might also be a little safer, as not all files have been converted to TS yet in remote clusters.